src/App.tsx

```diff
@@ -1,24 +1,244 @@
+  };
+
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  const changeStatus = (todo:Todo) => {
+    try {
+      setLoading(true);
+      setTodos(todos => todos.map(todoItem =>
+        todoItem.id === todo.id ? { ...todoItem, completed: !todoItem.completed } : todoItem
+      ));
+      
+    }
+     catch (error) {
+      setErrorMessage(Error.Loading);
+      setErrorMessage(Error.Updating)
+     }
+     finally {
+      setLoading(false);
+     }
+    }
+
+    
+  const changeTitle = (todo:Todo, editedTitle: string) => {
+    try {
+      setLoading(true);
+      setTodos(todos => todos.map(todoItem =>
+        todoItem.id === todo.id ? { ...todoItem, title: editedTitle } : todoItem
+      ));
+      
+    }
+     catch (error) {
+      setErrorMessage(Error.Loading);
+      setErrorMessage(Error.Updating)
+     }
+     finally {
+      setLoading(false);
+     }
+    }
+
+
+    function isAllTodosChecked(todos: Todo[]) {
+      for (const todo of todos) {
+        if (!todo.completed) {
+          return false;
+        }
+      }
+      return true;
+    }
+    
+    const changeStatusAllTodos = (todos: Todo[]) => {
+      try {
+        setLoading(true);
+  
+        setTodos(todos => todos.map((todoItem ) => todoItem.completed === false ? { ...todoItem, completed: true} : todoItem));
+
+        if(isAllTodosChecked(todos)) {
+          setTodos((todos) => todos.map(todoItem => ({ ...todoItem, completed: false })))
+        }
+       
+      }
+       catch (error) {
+        setErrorMessage(Error.Loading);
+        setErrorMessage(Error.Updating)
+       }
+       finally {
+        setLoading(false);
+       }
+    }
+
+  return ( 
+  <div className="todoapp">
+        <h1 className="todoapp__title">todos</h1>
+
+        <div className="todoapp__content">
+          <header className="todoapp__header">
+            {todos
+             && (
+               <button
+                 type="button"
+                 className={classNames('todoapp__toggle-all', {'active': !isAllTodosChecked})}
+                 onClick={() => changeStatusAllTodos(todos)}
+               />
+             )}
+
+            <form onSubmit={handleSubmit}>
+              <input
+                type="text"
+                className="todoapp__new-todo"
+                placeholder="What needs to be done?"
+                value={newTodoTitle}
+                onChange={handleInput}
+                disabled={loading}
+              />
+            </form>
+          </header>
 
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+          <section className="todoapp__main">
+            {{tempTodo} && (
+            <TodoList
+            todos={getFilteringTodos()} 
+            handleDelete={handleDelete} 
+            changeStatus={changeStatus}
+            changeTitle={changeTitle}
+            />)}
+          </section>
+          {todos.length !== 0 ? (
```

Comment:
use [conditional rendering](https://react.dev/learn/conditional-rendering)

Reply:
when I use conditional rendering, it doesn't work